### PR TITLE
Readme adjustments

### LINKED ISSUE #17: README improvements
I think having better instructions on how to setup the project helps the repo to get traction and is helpful specially for beginners.

What do you think @howardchung?

### PR DESCRIPTION
**Pull request by bdbch**
closes #17 

This PR adds a quick start section for the project and also includes an advanced setup section with instructions for the YouTube API key for now.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +26,23 @@ An website for watching videos together.
 - Text chat
 - Video chat
 
-## Configuration
+## Advanced Setup
+
+### YouTube API
+
+This project is using the YouTube API which means you will need to setup an API key. You can get one from Google [here](https://console.developers.google.com/?hl=de).
```
Comment (NotSimple) [inline README.md:33]: Link is to german version, should be https://console.developers.google.com/ without the ?hl=de

**Comment by NotSimple** [comment]
I recently found this project and haven't cloned the project yet. I will follow the quick start guide in this PR and see if I can get it working

File: README.md (hunk 1 above)
Comment (bdbch) [reply README.md:33]: Thanks for the catch. :)

**Comment by bdbch** [comment]
@NotSimple would be great! Thanks, let us know if something is missing!

File: README.md [hunk 2]
```diff
@@ -2,6 +2,16 @@
 
 An website for watching videos together.
 
+## Quick Start
+
+1. Clone this repo via `git clone git@github.com:howardchung/watchparty.git`
+2. Install npm dependencies for the project via `npm install`
+3. Start the server via `npm run dev`
```
Comment (howardchung) [inline README.md:9]: might want to add some notes on the PORT env var. Currently either the REACT_APP_SERVER_HOST env var needs to be set or client defaults to port 8080 on the same host.

File: README.md [hunk 3]
```diff
@@ -16,7 +26,23 @@ An website for watching videos together.
 - Text chat
 - Video chat
 
-## Configuration
+## Advanced Setup
+
+### YouTube API
+
+This project is using the YouTube API which means you will need to setup an API key. You can get one from Google [here](https://console.developers.google.com).
```
Comment (howardchung) [inline README.md:33]: Might want to mention this is actually optional, if YouTube isn't required

**Comment by NotSimple** [comment]
Ok so what I've found:
1. npm run dev (Step 3) fails on linux with latest node version
image
2. npm run dev (Step 3) works fine on linux with LTS version of node
3. npm run dev does not support windows
4. npm run start (Step 4) working on both windows and linux. I only tested using a youtube link to play a video so far.
5. npm run start (Step 4) takes a lot of time to start up the server
6. I haven't tested further, a suggestion outside the scope of this PR is implementing docker for higher portability and easier setup (specially for storage solutions, redis and/or postgres/other sql db)

**Comment by bdbch** [comment]
> Ok so what I've found:
> 
> 1. npm run dev (Step 3) fails on linux with latest node version
>    image
> 2. npm run dev (Step 3) works fine on linux with LTS version of node
> 3. npm run dev does not support windows
> 4. npm run start (Step 4) working on both windows and linux. I only tested using a youtube link to play a video so far.
> 5. npm run start (Step 4) takes a lot of time to start up the server
> 6. I haven't tested further, a suggestion outside the scope of this PR is implementing docker for higher portability and easier setup (specially for storage solutions, redis and/or postgres/other sql db) Opening #24 to propose it.

which version of node are you running? I'm on WSL2 using Ubutnu 18.04 and it works fine for me, never actually saw this problem.

Do you know what the error was on Windows?

File: README.md (hunk 3 above)
Comment (bdbch) [reply README.md:33]: True!

**Comment by NotSimple** [comment]
> which version of node are you running? I'm on WSL2 using Ubutnu 18.04 and it works fine for me, never actually saw this problem.
Do you know what the error was on Windows?

I'm also on WSL2 Ubuntu 18.04, using nvm. I tried Node version v14.3.0 and v14.0.0 and both show the error above. On Node v12.16.3 it works without a problem.

The windows error is:
image
fuser is a Unix-like command so it'll work on linux but not on windows and probably not osx either.

**Comment by howardchung** [comment]
Re: fuser I was encountering issues in the dev environment with the autorestart on file change not killing the server process, so I added that in front. We could maybe take it out (it was an issue when I was using nodemon but we've switched to ts-node-dev)

**Pull request merged**